### PR TITLE
[MD] Datasource Management - Create data source - Unit tests

### PR DESCRIPTION
**Pull request by mpabba3003**
### Description
Adding unit test cases for Create data source & other misc files in data source management
 
### Issues Resolved
https://github.com/opensearch-project/OpenSearch-Dashboards/issues/2019
https://github.com/opensearch-project/OpenSearch-Dashboards/issues/2020
 
### Check List
- [x] New functionality includes testing.
  - [ ] All tests pass
    - [ ] `yarn test:jest`
    - [ ] `yarn test:jest_integration`
    - [ ] `yarn test:ftr`
- [ ] New functionality has been documented.
- [ ] Commits are signed per the DCO using --signoff

### CODE REVIEW

**Review by zengyan-amazon — APPROVED** [review]

File: src/plugins/data_source_management/public/mocks.ts [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+      type: 'data-source',
+      description: 'test datasource2',
+      title: 'test',
+      get(field: string) {
+        const me: any = this || {};
+        return me[field];
+      },
+    },
+    {
+      id: 'alpha-test',
```
Comment (kristenTian) [inline src/plugins/data_source_management/public/mocks.ts:77]: just curious to know if the name means anything, alpha-test vs beta-test.  Also seems beta-test is been not used yet?

**Review by kristenTian — APPROVED** [review]

**Review by noCharger — APPROVED** [review]
Approved. Check for lint and add data-test-subj for interactive components like fields and buttons.

File: src/plugins/data_source_management/public/mocks.ts (hunk 1 above)
Comment (mpabba3003) [reply src/plugins/data_source_management/public/mocks.ts:77]: This is just mock data and just used for testing purposes .. Names mean nothing

Comment (kristenTian) [reply src/plugins/data_source_management/public/mocks.ts:77]: ok, feels a little confusing to me given it has state related names, if doesn't apply to our case maybe just keep consistency with previous naming.

**Pull request merged**